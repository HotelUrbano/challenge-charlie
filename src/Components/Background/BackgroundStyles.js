import styled from 'styled-components';

export const BackgroundImg = styled.div`
    width: 100%;
    height: 100%;
    overflow: auto;
`;