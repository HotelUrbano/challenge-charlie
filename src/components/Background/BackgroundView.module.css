.background{
  position: fixed;
  width: 100%;
  height: 100%;
  background-color: #121212;
  background-size: cover;
  background-position: center;
  top:0px;
  left:0px;
  z-index: -1;
}