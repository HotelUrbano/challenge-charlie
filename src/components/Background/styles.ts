import styled from 'styled-components'

export const Wrapper = styled.main`
  background: ${({ bg }) =>
    bg === '#06092b' ? '#06092b' : `url(https://www.bing.com${bg})`};
  background-size: cover;
  background-position: center;
  filter: ${({ bg }) => (bg === '#06092b' ? 'unset' : 'brightness(0.7)')};
  width: 100%;
  height: 100%;
  padding: 3rem;
  text-align: center;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`
