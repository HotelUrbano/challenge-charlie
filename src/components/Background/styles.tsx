import styled from 'styled-components';

interface BackgroundProps {
    bg?: any;
}
export const Container = styled.div`
    display: flex;
    align-items: center;
    justify-content: center;

    background-image: ${(props: BackgroundProps) =>
        props.bg ? `url(https://www.bing.com${props.bg})` : '#fffff'};

    width: 100%;
    height: 100vh;
`;
