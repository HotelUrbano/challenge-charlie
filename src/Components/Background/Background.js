import React, { Component } from 'react';
import regeneratorRuntime from "regenerator-runtime";
import { BackgroundImg } from './BackgroundStyles';

class Background extends Component {

    state = {
        image: null,
    };

    componentDidMount = async () => {
        const url = "https://cors-anywhere.herokuapp.com/https://www.bing.com/HPImageArchive.aspx?format=js&idx=0&n=1&mkt=pt-BR";
        const response = await fetch(url);
        const data = await response.json();
        this.setState({ image: data.images[0].url});
    }


	render() {

        const siteUrl = 'https://www.bing.com'

		return (
            <BackgroundImg className="main-bg" style={{ backgroundImage: `url(${ siteUrl + this.state.image })` }}>           
                {this.props.children}            
            </BackgroundImg>			
		);
	}
}
export default Background;