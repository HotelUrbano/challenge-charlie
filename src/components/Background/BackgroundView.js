import React from "react";
import styles from './BackgroundView.module.css'

const BackgroundView = ({url}) => (
  <div className={styles.background} style={{ backgroundImage: `url(${url})` }}></div>
);

export default BackgroundView;