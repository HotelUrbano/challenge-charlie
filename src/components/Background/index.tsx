import { useState, useEffect } from 'react'
import BingService from 'services/bingService'
import CacheService from 'services/cacheService'

import * as S from './styles'

interface IBackground {
  children: React.ReactNode
}

const Background = ({ children }: IBackground) => {
  const [bgImage, setBgImage] = useState('#06092b')

  const getBackgroundImage = async () => {
    const cache = await CacheService.getCacheData('background')
    if (cache) {
      console.log(cache)
      setBgImage(cache)
      return
    }
    const background = await BingService.getImageFromBing()
    CacheService.setCacheData('background', background, 3, 'HOURS')
    setBgImage(background)
  }

  useEffect(() => {
    getBackgroundImage()
  }, [])

  return <S.Wrapper bg={bgImage}>{children}</S.Wrapper>
}

export default Background
