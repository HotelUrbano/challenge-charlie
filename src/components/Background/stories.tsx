import { Story, Meta } from '@storybook/react/types-6-0'
import Background from '.'

export default {
  title: 'Background',
  component: Background
} as Meta

export const Default: Story = () => <Background />
