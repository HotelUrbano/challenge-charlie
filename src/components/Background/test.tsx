import { render, screen } from '@testing-library/react'

import Background from '.'

describe('<Background />', () => {
  it('should render the background', () => {
    const { container } = render(<Background>hi</Background>)

    expect(
      screen.getByRole('heading', { name: /Background/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
