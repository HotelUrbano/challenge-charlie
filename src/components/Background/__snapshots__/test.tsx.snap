// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Background /> should render the heading 1`] = `
<main
  class=""
>
  <h1>
    Background
  </h1>
</main>
`;
